File: packages/typescript/src/infrastructure/infraCheck.ts
import axios from 'axios';

export async function checkOllamaStatus(baseUrl: string = 'http://localhost:11434'): Promise<boolean> {
  try {
    const response = await axios.get(`${baseUrl}/api/tags`);
    return response.status === 200;
  } catch (error) {
    return false;
  }
}

File: packages/python/llm_local_client/infrastructure/infra_check.py
import requests

def check_ollama_status(base_url: str = 'http://localhost:11434') -> bool:
    try:
        response = requests.get(f"{base_url}/api/tags", timeout=5)
        return response.status_code == 200
    except requests.RequestException:
        return False
